core/controllers/topics_and_skills_dashboard.py

```diff
@@ -174,6 +177,16 @@ def post(self):
             state_domain.SubtitledHtml.from_dict(explanation_dict))
         skill_services.save_new_skill(self.user_id, skill)
 
+        for filename in filenames:
+            image = self.request.get(filename)
```

Comment:
what of the request doesn't have filename key, should we raise exception?

Reply:
The images are added alongwith the filenames, so if one is there, the other would be there. Should an exception be thrown here?

Reply:
Yeah I think we should raise 400 instead unexpected 500

Reply:
I think 400, but you could still log an error (logging.error). Check with @kevintab95 what is needed for it to show in StackDriver.

Reply:
It depends on how you're handling the error in the UI. For the error to show up on StackDriver, you can either (a) not handle the failure case in the AJAX call in which case the http interceptor will log the error, or (b) handle the AJAX failure and throw an explicit error with a more descriptive message if required.

Reply:
The `image_validation_service.validate_image_and_filename()` function does take care of this as if this is not declared, 'image' would be None, which is explicitly checked in the function.
So, this would case (a) as @kevintab95 mentioned, right? (not handle failure case in the frontend, but the backend throws the error with message)

Reply:
Yep, adding a logging.error should suffice in that case.

Reply:
Done, added logging.error.